pkg/controller/signup.go

```diff
@@ -46,6 +52,85 @@ func (s *Signup) PostHandler(ctx *gin.Context) {
 	ctx.Writer.WriteHeaderNow()
 }
 
+// UpdateVerificationHandler starts the verification process and updates a usersignup resource. The
+// ctx should contain a JSON body in the request with a country_code and a phone_number string
+func (s *Signup) UpdateVerificationHandler(ctx *gin.Context) {
+	userID := ctx.GetString(context.SubKey)
+	signup, err := s.signupService.GetUserSignup(userID)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Error(ctx, err, "usersignup not found")
+			errors.AbortWithError(ctx, http.StatusNotFound, err, "usersignup not found")
+			return
+		}
+		log.Error(ctx, err, "error retrieving usersignup")
+		errors.AbortWithError(ctx, http.StatusInternalServerError, err, fmt.Sprintf("error retrieving usersignup: %s", userID))
+		return
+	}
+
+	// check that verification is required before proceeding
+	if signup.Spec.VerificationRequired == false {
+		log.Errorf(ctx, errors.NewForbiddenError("forbidden request", "verification code will not be sent"), "phone verification not required for usersignup: %s", userID)
+		ctx.AbortWithStatus(http.StatusBadRequest)
+		return
+	}
+
+	// Read the Body content
+	var phone Phone
+	if err := ctx.BindJSON(&phone); err != nil {
+		log.Errorf(ctx, err, "request body does not contain required fields phone_number and country_code")
+		errors.AbortWithError(ctx, http.StatusBadRequest, err, "error reading request body")
+		return
+	}
+
+	// normalize phone number
+	r := strings.NewReplacer("(", "",
+		")", "",
+		" ", "",
+		"-", "")
+	countryCode := r.Replace(phone.CountryCode)
+	phoneNumber := r.Replace(phone.PhoneNumber)
```

Comment:
Please add tests for different valid cases like `(123)`, `1 2 3`, etc. As far as I can see you have only tests for the invalid symbols if I'm not missing anything.

Reply:
updated here: 3cd7858e9204ccb4210d579bd1d341567ee9a2b0